internal/provider/resource_saml_idp.go

```diff
@@ -40,18 +45,18 @@ func resourceWizSAMLIdP() *schema.Resource {
 			},
 			"use_provider_managed_roles": {
 				Type:        schema.TypeBool,
-				Description: "Use provider managed roles?",
-				Optional:    true,
-				Default:     false,
+				Description: "Use provider (IdP) managed roles? This attribute needs to be set to accomodate for update lifecycle operations.",
+				Required:    true,
 			},
 			"allow_manual_role_override": {
 				Type:        schema.TypeBool,
-				Description: "Allow manual override for role assignment? Must be set `true` if `use_provided_roles` is false.",
-				Optional:    true,
-				Default:     true,
-				RequiredWith: []string{
-					"use_provider_managed_roles",
-				},
+				Required:    true,
+				Description: "Allow manual override for role assignment? This attribute must be set to `true` if `use_provider_managed_roles` attribute is set to `false`. This field needs to be set to accomodate for update lifecycle operations.",
+			},
+			"merge_groups_mapping_by_role": {
+				Type:        schema.TypeBool,
+				Description: "Manage group mapping by role? This attribute needs to be set to accomodate for update lifecycle operations.",
+				Required:    true,
```

Comment:
Why was this changed from optional to required?  The GraphQL schema indicates that this is optional.

```
input CreateSAMLIdentityProviderInput {
  name: String!
  issuerURL: String
  loginURL: String!
  logoutURL: String
  useProviderManagedRoles: Boolean!
  allowManualRoleOverride: Boolean
  certificate: String!
  domains: [String!]
  groupMapping: [SAMLGroupMappingCreateInput!]
  mergeGroupsMappingByRole: Boolean
}
```

Reply:
see https://github.com/AxtonGrams/terraform-provider-wiz/pull/42#discussion_r1036255152